### PR TITLE
Improve some comments in GenDialect.cpp

### PR DESCRIPTION
**Pull request by tsymalla**
I corrected some error messages which could be misleading.

### CODE REVIEW

File: lib/TableGen/GenDialect.cpp [hunk 1]
```diff
@@ -571,7 +571,9 @@ class SymbolTable {
   std::unordered_set<std::string> m_names;
 
 public:
-  std::string chooseName(StringRef name) { return chooseName(ArrayRef(name)); }
+  std::string chooseName(StringRef name) {
+    return chooseName(makeArrayRef(name));
+  }
```
Comment (Flakebi) [inline lib/TableGen/GenDialect.cpp:576]: Why did you revert this change?

Comment (tsymalla) [reply lib/TableGen/GenDialect.cpp:576]: I did not revert this intentionally. I just used the GitHub web editor. Maybe the fork being generated is not up-to-date.

Comment (Flakebi) [reply lib/TableGen/GenDialect.cpp:576]: Sounds like a race-condition in GitHub’s web ui. (I.e. start editing from a certain state, but once one presses save, GitHub gets the new state, which is different when someone pushed in-between.)

Comment (tsymalla) [reply lib/TableGen/GenDialect.cpp]: Yes. Sounds likely, and I rebased onto recent dev.

**Review by nhaehnle — APPROVED** [review]
Thanks!

**Comment by tsymalla** [comment]
@nhaehnle You're welcome. I guess it is currently you taking care of merging PRs?
